examples/06_rgb_full_resolution_saver.py

```diff
@@ -5,42 +5,53 @@
 
 import cv2
 import depthai as dai
-import numpy as np
 
 # Start defining a pipeline
 pipeline = dai.Pipeline()
 
 # Define a source - color camera
 cam_rgb = pipeline.createColorCamera()
-cam_rgb.setPreviewSize(3840, 2160)
 cam_rgb.setResolution(dai.ColorCameraProperties.SensorResolution.THE_4_K)
-cam_rgb.setInterleaved(False)
 
-# Create output
+# Create RGB output
 xout_rgb = pipeline.createXLinkOut()
 xout_rgb.setStreamName("rgb")
-cam_rgb.preview.link(xout_rgb.input)
+cam_rgb.video.link(xout_rgb.input)
+
+# Create encoder to produce JPEG images
+video_enc = pipeline.createVideoEncoder()
+video_enc.setDefaultProfilePreset(cam_rgb.getVideoSize(), cam_rgb.getFps(), dai.VideoEncoderProperties.Profile.MJPEG)
+cam_rgb.video.link(video_enc.input)
+
+# Create JPEG output
+xout_jpeg = pipeline.createXLinkOut()
+xout_jpeg.setStreamName("jpeg")
+video_enc.bitstream.link(xout_jpeg.input)
 
 # Pipeline defined, now the device is assigned and pipeline is started
 device = dai.Device(pipeline)
 device.startPipeline()
 
 # Output queue will be used to get the rgb frames from the output defined above
-q_rgb = device.getOutputQueue(name="rgb", maxSize=4, blocking=False)
+q_rgb = device.getOutputQueue(name="rgb", maxSize=30, blocking=False)
+q_jpeg = device.getOutputQueue(name="jpeg", maxSize=30, blocking=True)
 
 # Make sure the destination path is present before starting to store the examples
 Path('06_data').mkdir(parents=True, exist_ok=True)
 
 while True:
-    in_rgb = q_rgb.get()  # blocking call, will wait until a new data has arrived
-    # data is originally represented as a flat 1D array, it needs to be converted into HxWxC form
-    shape = (3, in_rgb.getHeight(), in_rgb.getWidth())
-    frame_rgb = in_rgb.getData().reshape(shape).transpose(1, 2, 0).astype(np.uint8)
-    frame_rgb = np.ascontiguousarray(frame_rgb)
-    # frame is transformed and ready to be shown
-    cv2.imshow("rgb", frame_rgb)
-    # after showing the frame, it's being stored inside a target directory as a PNG image
-    cv2.imwrite(f"06_data/{int(time.time() * 10000)}.png", frame_rgb)
+    in_rgb = q_rgb.tryGet()  # non-blocking call, will return a new data that has arrived or None otherwise
+    print(in_rgb)
```

Comment:
remove debug print, other than that LGTM

Reply:
I'll wait with this example until `gen2_improvements` branch is merged - as @themarpe suggested, I tried to use VideoEncoder and I've detected a bug with it - but also I was able to see the huge performance gain with VideoEncoder on device side, so I think it's worth waiting

Reply:
@VanDavv `gen2_improvements` was merged into `gen2_develop` - feel free to update.